File: tictactoe.c
#include<stdio.h>
void display(char a[3][3]) {
	printf("\n");
	for(int i = 0; i < 3; i++) {
        for(int j = 0; j < 3; j++) {
    	    printf("%c  ",a[i][j]);
        }
        printf("\n");
    }
	printf("\n");
}
int checkResult(char a[3][3]) {
	char ch = '_';
	if(a[0][0] != '_' && a[0][0] == a[1][1] && a[1][1] == a[2][2]) {
		ch = a[0][0];
	} else if(a[0][2] != '_' && a[0][2] == a[1][1] && a[1][1] == a[2][0]) {
		ch = a[0][2];
	} else {
		for (int i = 0; i < 3; i++) {
			if (a[i][0] != '_' && a[i][0] == a[i][1] && a[i][1] == a[i][2]) {
				ch = a[i][0];
				break;
			} else if (a[0][i] != '_' && a[0][i] == a[1][i] && a[1][i] == a[2][i]) {
				ch = a[0][i];
				break;
			}
		}
	}
	if(ch == 'x') {
		printf("Player 1 has won\n\n");
		return 1;
	} else if(ch == 'o') {
		printf("Player 2 has won\n\n");
		return 1;
	} else {
		return 0;
	}
}
int input(char a[3][3], int i) {
	int n,m;
	while(1) {
		if(i % 2 != 0) {
			printf("Player 1 chance: ");
		} else {
			printf("Player 2 chance: ");
		}
		scanf("%d %d",&n,&m);
		if((n >= 1 && n <= 3) && (m >= 1 && m <= 3)) {
			if(a[n-1][m-1] == '_') {
				break;
			}
		} 
		printf("Wrong Input Retry\n\n");
	}
	if(i % 2 != 0) {
		a[n-1][m-1] = 'x';
	} else {
		a[n-1][m-1] = 'o';
	}
	display(a);
	return checkResult(a);
}
int main() {
	char a[3][3];
	printf("For putting the x or o in your desired location you must enter the coordinates of the point. Select from the given below.\n");
	printf("(1 1) (1 2) (1 3) (2 1) (2 2) (2 3) (3 1) (3 2) (3 3)\n");
	for(int i = 0; i < 3; i++) {
		for(int j = 0; j < 3; j++) {
			a[i][j] = '_';
		}
	}
	printf("\n");
	display(a);
    printf("\n");
    for(int i = 1; i <= 9; i++) {
        if(input(a,i))
			return 0;
    }
    printf("Match is draw\n\n");
    return 0;
}

File: tictactoe.py
def display(board):
    print()
    for row in board:
        print("  ".join(row))
    print()

def result(board):
    winner = '_'
    if board[0][0] != '_' and board[0][0] == board[1][1] == board[2][2]:
        winner = board[0][0]
    elif board[0][2] != '_' and board[0][2] == board[1][1] == board[2][0]:
        winner = board[0][2]
    else:
        for i in range(3):
            if board[i][0] != '_' and board[i][0] == board[i][1] == board[i][2]:
                winner = board[i][0]
                break
            elif board[0][i] != '_' and board[0][i] == board[1][i] == board[2][i]:
                winner = board[0][i]
                break
    if winner == 'x':
        print("Player 1 has won")
        return 1
    elif winner == 'o':
        print("Player 2 has won")
        return 1
    else:
        return 0

def player_input(board, turn):  # Renamed from 'input' to 'player_input'
    while True:
        if turn % 2 != 0:
            print("Player 1 chance: ", end="")
        else:
            print("Player 2 chance: ", end="")
        
        try:
            n, m = map(int, input().split())
            if 1 <= n <= 3 and 1 <= m <= 3:  
                if board[n - 1][m - 1] == '_': 
                    break
            print("Wrong Input, Retry")
            print()
        except ValueError:
            print("Invalid Input, Retry")
    
    if turn % 2 != 0:
        board[n - 1][m - 1] = 'x'
    else:
        board[n - 1][m - 1] = 'o'
    
    display(board)
    return result(board)

def game():
    board = [['_' for _ in range(3)] for _ in range(3)]
    print()
    print("For putting the x or o in your desired location you must enter the coordinates of the point. Select from the given below.")
    print("(1 1) (1 2) (1 3) (2 1) (2 2) (2 3) (3 1) (3 2) (3 3)")
    display(board)
    for i in range(1,9):
        if player_input(board, i):  
            return
    print("Match is draw")
    return

game()